src/librustc/infer/higher_ranked/mod.rs

```diff
@@ -520,6 +550,56 @@ pub fn leak_check<'a,'tcx>(infcx: &InferCtxt<'a,'tcx>,
             return Err((skol_br, tainted_region));
         }
     }
+
+    for (_, &skol) in skol_map {
+        // The outputs from a skolemized variable must all be
+        // equatable with `'static`.
+        let outgoing_taints = infcx.tainted_regions(snapshot,
+                                                    skol,
+                                                    TaintDirections::outgoing());
+        for &tainted_region in &outgoing_taints {
+            match tainted_region {
+                ty::ReVar(vid) if new_vars.contains(&vid) => {
+                    // Some region variable that doesn't escape
```

Comment:
Note for @nikomatsakis: we spent a while talking about this during our video review. The summary follows.

There are two possible cases here. 

- `tainted_region` eventually reaches a skolemized variable, which *must* be `skol` (because otherwise we would have already returned `Err`). In that case, `tainted_region` be `skol`.

- `tainted_region` never reaches a skolemized variable. In that case, we can safely choose `'static` as an upper bound incoming edges. This is a conservative choice -- the LUB might be one of the incoming skolemized variables, which we might know by ambient bounds. We can consider a more clever choice of upper bound later (modulo some theoretical breakage).